File: leetcode_python/samples/example/_1768_Merge_String_Alternatively.py
def mergealternately(word1: str, word2: str) -> str:
    p1 = 0
    p2 = 0

    l1 = len(word1)
    l2 = len(word2)

    result = ""
    while (p1 < l1) or (p2 < l2):
        if p1 < l1:
            result += word1[p1]
            p1 += 1
        if p2 < l2:
            result += word2[p2]
            p2 += 1

    return result

File: leetcode_java/src/main/java/org/example/_1768_Merge_String_Alternatively.java
package org.example;

public class _1768_Merge_String_Alternatively {


    public static String mergeAlternately(String word1, String word2) {

        int p1 = 0;
        int p2 = 0;

        int l1 = word1.length();
        int l2 = word2.length();
        StringBuilder builder = new StringBuilder();

        while (p1 < l1 || p2 < l2) {

            if (p1 < l1) {
                builder.append(word1.charAt(p1));
                p1++;
            }
            if (p2 < l2) {
                builder.append(word2.charAt(p2));
                p2++;
            }
        }

        return builder.toString();
    }


}
